presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -191,6 +188,194 @@ This table can then be queried in Presto::
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:** Limit of number of splits waiting to be served by split source. After reaching this limit writers will stop writing new splits to split source until some of them are used by workers. Higher value will increase memory usage, but will allow to concentrate all IO at one time which may be much faster and increase resources utilization.
+
+
+``hive.max-partitions-per-writers``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:** Maximum number of partitions per writer. If higher number of partitions per writer will be required to complete query, the query will fail. By manipulating this value one may change how large queries are meant to be dropped from DB which may help with error detection.
+
+
+``hive.max-split-iterator-threads``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:** This property describes how many threads may be used to iterate through splits when loading them to the worker nodes. Higher value may increase parallelism, but high concurrency may cause time being wasted on context switching.
+
+
+``hive.max-split-size``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``64 MB``
+ * **Description:** This value describes max size of split that is created after using all ``hive-max-initial-split-size`` of initial splits. The logic of initial splits is described in ``hive.max-initial-splits``. Having this value higher causes smaller parallelism which may be desirable when queries are very large and cluster is stable allowing to process data locally more efficiently without wasting time for context switching, synchronization and data collecting. The optimal value should be aligned with average query size in system.
+
+
+``hive.metastore.partition-batch-size.max``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:** This together with ``hive.metastore.partition-batch-size.min`` defines range of partition sizes read from Hive. First partition is always of size ``hive.metastore.partition-batch-size.min`` and each following partition is two times bigger then previous up to ``hive.mestastore.partition-batch-size.max`` (the formula for ``n`` partition size is min(``hive.metastore.partition-batch-size.max``, (``2``^``n``) * ``hive.metastore.partition-batch-size.min``)). This algorithm allows to adjust partition size live to what is required. If size of queries in system differs siginificantly, then this range should be extended to better adjust to processed case. In case of cluster working with queries with about the same size, both values may be same for maximal attunement giving slight edge in processing time.
```

Comment:
s/defines range of partition sizes/defines the range of partition sizes/
s/First partition/The first partition
s/bigger then/bigger than
s/``n`` partition size/partition size ``n``
s/This algorithm allows to adjust partition size live to what is required. If size of queries in system differs siginificantly, then this range should be extended to better adjust to processed case. In case of cluster working with queries with about the same size, both values may be same for maximal attunement giving slight edge in processing time./This algorithm allows for live adjustment of partition size according to the processing requirements.  If the queries in the system will differ significantly from each other in size, then this range should be extended to better adjust to processing requirements.  If the queries in the system will mostly be of the same size, then setting both values to the same maximally tuned value may give a slight edge in processing time.